lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1460,6 +1460,9 @@ static bool ParseClangImporterArgs(ClangImporterOptions &Opts,
     Opts.PCHDisableValidation |= Args.hasArg(OPT_pch_disable_validation);
   }
 
+  if (FrontendOpts.DisableImplicitModules)
```

Comment:
This is an important change, it essentially prohibits implicit loading of Clang modules when Swift explicit module loading is enabled. It is not clear why such a combination would be useful, and it is not something we intend to validate, so may as well make it not possible.